torch/_functorch/_aot_autograd/fx_utils.py

```diff
@@ -0,0 +1,307 @@
+        "FQN to.  It's not clear what you want the API to do in this case: do you want to "
+        "instead return a struct of nodes showing how to assemble the subclass?  But you "
+        "don't (directly) have the metadata for the subclass?  If you have a concrete use "
+        "case, please file an issue so we can understand it and design an API that works for your case."
+    )
+
+
+def get_named_param_nodes(graph: fx.Graph) -> dict[str, fx.Node]:
+    """Get parameter nodes mapped by their fully qualified names.
+
+    This function traverses the graph to find all parameter input nodes and
+    returns them in a dictionary where keys are the parameter names (FQNs)
+    and values are the corresponding FX nodes.
+
+    Args:
+        graph: The FX joint graph with descriptors
+
+    Returns:
+        A dictionary mapping parameter names (str) to their corresponding FX nodes.
+
+    Raises:
+        RuntimeError: If subclass tensors are encountered (not yet supported), as
+        with subclasses a FQN does not necessarily map to a single plain tensor.
+    """
+    r = {}
+    for n in graph.nodes:
+        if n.op == "placeholder":
+            desc = n.meta["desc"]
+            if isinstance(desc, SubclassGetAttrAOTInput):
+                _raise_fqn_subclass_not_implemented()
+            elif isinstance(desc, ParamAOTInput):
+                r[desc.target] = n
+    return r
+
+
+def get_named_buffer_nodes(graph: fx.Graph) -> dict[str, fx.Node]:
+    """Get buffer nodes mapped by their fully qualified names.
+
+    This function traverses the graph to find all buffer input nodes and
+    returns them in a dictionary where keys are the buffer names (FQNs)
+    and values are the corresponding FX nodes.
+
+    Args:
+        graph: The FX joint graph with descriptors
+
+    Returns:
+        A dictionary mapping buffer names (str) to their corresponding FX nodes.
+
+    Raises:
+        RuntimeError: If subclass tensors are encountered (not yet supported), as
+        with subclasses a FQN does not necessarily map to a single plain tensor.
+    """
+    r = {}
+    for n in graph.nodes:
+        if n.op == "placeholder":
+            desc = n.meta["desc"]
+            if isinstance(desc, SubclassGetAttrAOTInput):
+                _raise_fqn_subclass_not_implemented()
+            elif isinstance(desc, BufferAOTInput):
+                r[desc.target] = n
+    return r
+
+
+def get_param_nodes(graph: fx.Graph) -> list[fx.Node]:
+    """Get all parameter nodes from a graph as a list.
+
+    You can rely on this providing the correct order of parameters you need
+    to feed into the joint graph (at the very beginning of the argument list,
+    before buffers).
+
+    Args:
+        graph: The FX joint graph with descriptors
+
+    Returns:
+        A list of FX nodes representing all parameters in the graph.
+
+    Raises:
+        RuntimeError: If subclass tensors are encountered (not yet supported), as
```

Comment:
i could imagine someone arguing for this function just returning the flat list of parameters.  But i'm ok with leaving it raising for now until someone complains

Reply:
I actually don't understand this comment. Isn't this a flat list of parameter (nodes)? You can't return the parameters, I don't have 'em!